driver-core/src/main/com/mongodb/client/model/Windows.java

```diff
@@ -85,12 +87,52 @@
  */
 @Beta
 public final class Windows {
+    /**
+     * Creates a window from a document in situations when there is no builder method that better satisfies your needs.
+     * This method cannot be used to validate the document syntax.
+     * <p>
+     * <i>Example</i><br>
+     * The following code creates two functionally equivalent windows, though they may not be {@linkplain #equals(Object) equal}.
+     * <pre>{@code
+     *  Window pastWeek1 = Windows.timeRange(-1, MongoTimeUnit.WEEK, Windows.Bound.CURRENT);
+     *  Window pastWeek2 = Windows.of(
+     *          new Document("range", Arrays.asList(-1, "current"))
+     *                  .append("unit", "week"));
+     * }</pre>
+     *
+     * @param window A document representing the required window.
+     * @return The constructed window.
+     */
+    public static Window of(final Document window) {
+        return new DocumentWindow(notNull("window", window));
+    }
+
+    /**
+     * Creates a window from a document in situations when there is no builder method that better satisfies your needs.
+     * This method cannot be used to validate the document syntax.
+     * <p>
+     * <i>Example</i><br>
+     * The following code creates two functionally identical windows, though they may not be {@linkplain #equals(Object) equal}.
+     * <pre>{@code
+     *  Window pastWeek1 = Windows.timeRange(-1, MongoTimeUnit.WEEK, Windows.Bound.CURRENT);
+     *  Window pastWeek2 = Windows.of(
+     *          new BsonDocument("range", new BsonArray(Arrays.asList(new BsonInt32(-1), new BsonString("current"))))
+     *                  .append("unit", new BsonString("week")));
+     * }</pre>
+     *
+     * @param window A document representing the required window.
+     * @return The constructed window.
+     */
+    public static Window of(final BsonDocument window) {
```

Comment:
I think you could just have a single `of` method with a `window` parameter of type `Bson`.  This would correspond more closely to the rest of the driver API.

Reply:
Indeed. Done.